recipes/src/main/java/io/quarkus/updates/core/quarkus39/RemoveMavenCompilerAnnotationProcessor.java

```diff
@@ -0,0 +1,84 @@
+package io.quarkus.updates.core.quarkus39;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.maven.MavenVisitor;
+import org.openrewrite.xml.FilterTagChildrenVisitor;
+import org.openrewrite.xml.tree.Xml;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class RemoveMavenCompilerAnnotationProcessor extends Recipe {
+
+    @Option(displayName = "Annotation processor groupId", description = "The groupId of the annotation processor.", example = "org.hibernate.orm", required = true)
+    String groupId;
+
+    @Option(displayName = "Annotation processor artifactId", description = "The artifactId of the annotation processor.", example = "hibernate-jpamodelgen", required = true)
+    String artifactId;
+
+    @Option(displayName = "Annotation processor fully-qualified class", description = "The fully-qualified class of the annotation processor.", example = "com.example.MyProcessor", required = false)
+    String processorClass;
+
+    @Override
+    public String getDisplayName() {
+        return "Remove an annotation processor from the Maven Compiler plugin configuration";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Remove an annotation processor from the Maven Compiler plugin configuration.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new MavenVisitor<ExecutionContext>() {
+            @Override
+            public Xml visitTag(Xml.Tag tag, ExecutionContext ctx) {
+                Xml.Tag mavenCompilerPluginTag = (Xml.Tag) super.visitTag(tag, ctx);
+
+                if (isPluginTag("org.apache.maven.plugins", "maven-compiler-plugin")) {
+                    Optional<Xml.Tag> configuration = mavenCompilerPluginTag.getChild("configuration");
+
+                    if (configuration.isPresent()) {
+                        Optional<Xml.Tag> annotationProcessorPathsWrapper = configuration.get()
+                                .getChild("annotationProcessorPaths");
+                        if (annotationProcessorPathsWrapper.isPresent()) {
+                            List<Xml.Tag> annotationProcessorPaths = annotationProcessorPathsWrapper.get().getChildren();
+                            String childName = annotationProcessorPaths.size() > 0 ? annotationProcessorPaths.get(0).getName()
+                                    : "path";
+
+                            mavenCompilerPluginTag = FilterTagChildrenVisitor.filterTagChildren(mavenCompilerPluginTag, annotationProcessorPathsWrapper.get(), 
+                            			child ->
+                            			!(child.getName().equals(childName)
```

Comment:
It's minor but you don't need to have this logic about `childName`. Whatever name it is there, it should be an annotation processor path (or things are invalid).
Also, you don't need to handle the `size() > 0` case as if there are no children, we don't need to filter them.

Reply:
I wasn't sure how deep the filter went, if it was for first-level or multi-level, but thanks.